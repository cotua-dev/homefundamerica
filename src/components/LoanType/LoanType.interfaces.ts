export interface Props {
    title: string;
    content: string;
    learnMore: string;
}
