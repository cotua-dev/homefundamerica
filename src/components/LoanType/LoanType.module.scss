section#loanTypeSection {
    height: 40vh;

    @media screen and (max-width: 950px) {
        height: 100%;
    }
}
