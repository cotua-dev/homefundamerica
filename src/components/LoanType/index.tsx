import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import { Props } from './LoanType.interfaces';
import * as styles from './LoanType.module.scss';
import * as sharedStyles from '../sharedStyles/shared.module.scss';

export const LoanType: FunctionComponent<Props> = (props: Props): ReactElement<Props> => {
    return (
        <section id={styles.loanTypeSection} className={sharedStyles.loanPositive}>
            <div className={sharedStyles.contentWrapper}>
                <h2>{props.title}</h2>
                <p>{props.content}</p>
                <p>
                    {props.learnMore}
                    <a href="tel:8888186393">888-818-6393</a>
                </p>
            </div>
        </section>
    );
}
