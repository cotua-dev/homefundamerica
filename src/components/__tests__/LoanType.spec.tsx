import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { LoanType } from '../LoanType';

describe('LoanType Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(
            <LoanType
                title="Hello World"
                content="This is test content"
                learnMore="Learn more about something"
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
