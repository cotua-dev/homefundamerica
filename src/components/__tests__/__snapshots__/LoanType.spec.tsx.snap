// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanType Component renders 1`] = `
<section>
  <div>
    <h2>
      Hello World
    </h2>
    <p>
      This is test content
    </p>
    <p>
      Learn more about something
      <a
        href="tel:8888186393"
      >
        888-818-6393
      </a>
    </p>
  </div>
</section>
`;
